### ISSUE TITLE
dsContributor shows up in list role assignments result but not in user interface

### ISSUE THREAD

**Issue by janvanmansum** [IQSS/dataverse#8929]
**What steps does it take to reproduce the issue?**
This assumes your dataverse collection is configured to allow all authenticated users to create datasets.

Choose a dataset and retrieve the role assignments on that dataset through the API. Note that the following role assignment is included:

```json
 {
   "id": 12345,
   "assignee": ":authenticated-users",
   "roleId": 5,
   "_roleAlias": "dsContributor",
   "definitionPointId": 1
 }
```

Note also that the only permission in this role is AddDataset, which is only useful for dataverse collections, as you cannot create a dataset within a dataset.

Now for comparison:

1. Log in as admin
2. Open the same dataset
3. Go to Edit Dataset -> Permissions -> Dataset
4. The role assignment ":authenticated-users" -> "dsContributor" is not included

* What did you expect to happen?
I should expect the role assignment to be included in both the API result and the UI or in neither. Since the role assignment has no practical effect on the dataset that inherits it, it is probably best not to include it.

**Which version of Dataverse are you using?**
5.10.1

**Any related open or closed issues to this bug report?**
N/a

**Comment by pdurbin** [comment]
@janvanmansum yes, I agree this confusing and buggy. Here's a dataset permissions screenshot from me testing on the develop branch (post 5.11.1):

![Screen Shot 2022-09-12 at 4 39 28 PM](https://user-images.githubusercontent.com/21006/189754349-880284cc-db41-4fd1-8382-ac19784330da.png)

I created the parent dataverse and dataset with the dataverseAdmin user.

In the curl output below listing dataset role assignments, here are the definitions points:

- definitionPointId 13: the dataset
- definitionPointId 12: the parent collection

As shown in the screenshot, the "Admin" role is inherited from the parent collection (definitionPointId 12).

The "Contributor" role, defined at the dataset (definitionPointId 13) is given to dataverseAdmin as part of that user creating the dataset.

As you say, dsContributor has only AddDataset permission...

```
$ cat scripts/api/data/role-dsContributor.json 
{
  "alias": "dsContributor",
  "name": "Dataset Creator",
  "description": "A person who can add datasets within a dataverse.",
  "permissions": [
    "AddDataset"
  ]
}
```

... and this role really has nothing to do with datasets.

As you say, `:authenticated-users` having dsContributor does not appear in the UI. This probably makes sense. What doesn't make sense is why the this role is assigned at the dataset. To further repeat what you've already told us, you can't add a dataset to a dataset. 😄 

I'd suggest (to whoever wants to work on this) investigating why the dsContributor role is being assigned at the dataset level, and, if there's no reason for it, to stop assigning it. That way it won't appear in the API output and we'll have consistency with what the UI is showing.

Phew! Thanks for the bug report! 😄 

```
$ curl -s -H X-Dataverse-key:$API_TOKEN $SERVER_URL/api/datasets/$ID/assignments | jq .
{
  "status": "OK",
  "data": [
    {
      "id": 15,
      "assignee": "@dataverseAdmin",
      "roleId": 6,
      "_roleAlias": "contributor",
      "definitionPointId": 13
    },
    {
      "id": 13,
      "assignee": "@dataverseAdmin",
      "roleId": 1,
      "_roleAlias": "admin",
      "definitionPointId": 12
    },
    {
      "id": 14,
      "assignee": ":authenticated-users",
      "roleId": 5,
      "_roleAlias": "dsContributor",
      "definitionPointId": 12
    }
  ]
}
```

**Comment by scolapasta** [comment]
I don't beleive the role is being assigned at the dataset level - if you see the definition point is 1 (which is the root dataverse in most / all installations), so this is just being inherited from the dataverse. Which is fine, of course, so we should just do in the API what we do in the UI and suppress these types of inherited roles. That said, the fact that they are different means the suppression must be happening in the UI code, and could / should probably be at a lower level where it's shared by both.
